addons/l10n_ro_efactura/models/ciusro_document.py

```diff
@@ -164,19 +164,26 @@ def _request_ciusro_download_answer(self, company, key_download, session):
 
         # E-Factura gives download response in ZIP format
         zip_ref = zipfile.ZipFile(io.BytesIO(result['content']))
-        signature_file = next(file for file in zip_ref.namelist() if 'semnatura' in file)
+        # The ZIP will contain two files, one with the original invoice or with the identified errors (as the case may be)
+        # and the other with the electronic signature (containing 'semnatura').
+        # If there is an error (status == 'nok') we want to provide the file with the errors.
+        if status == 'nok':
+            signature_file = next(file for file in zip_ref.namelist() if 'semnatura' not in file)
+        else:
+            signature_file = next(file for file in zip_ref.namelist() if 'semnatura' in file)
         xml_bytes = zip_ref.open(signature_file)
         root = etree.parse(xml_bytes)
-        error_element = root.find('.//ns:Error', namespaces=NS_HEADER)
-        if error_element is not None:
-            return {'error': error_element.get('errorMessage')}
+        error_elements = root.findall('.//ns:Error', namespaces=NS_HEADER)
+        if error_elements:
+            error_message = ('\n\n').join(error.get('errorMessage') for error in error_elements)
 
         # Pretty-print the XML content of the signature file to be saved as attachment
         attachment_raw = etree.tostring(root, pretty_print=True, xml_declaration=True, encoding='UTF-8')
         return {
             'attachment_raw': attachment_raw,
             'key_signature': root.findtext('.//ns:SignatureValue', namespaces=NS_SIGNATURE),
             'key_certificate': root.findtext('.//ns:X509Certificate', namespaces=NS_SIGNATURE),
+            'error': error_message if error_elements else False,
```

Comment:
need to adapt the docstring https://github.com/odoo/odoo/pull/190618/files#diff-97dcc436b83588a0003e5b60f1914f3b8fd79bd0deda3e57f4729746240abbcdR153